yext/answers-hitchhiker-theme
# Map issues fixed for RTL locale/language

Fix Mapbox issues when using RTL styling and language

- Map not showing up in rtl format issue is from Mapbox only (Google works on all map pages) due to[ css issues](https://github.com/mapbox/mapbox-gl-js/issues/9329) from their api. This is fixed in [v1.9.0](https://github.com/mapbox/mapbox-gl-js/blob/main/CHANGELOG.md#190). This pr updated the version to v1.13.1 only since they stop supporting ie11 in higher version.
- Use [MapBoxLanguage](https://github.com/mapbox/mapbox-gl-language) plugin to update language on displayed on map. Previously, map will display in english only.
- MapBox by default doesn't support RTL languages. Updated to use [mapbox-rtl-text](https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-rtl-text/) plugin to resolve the issue.
- Update zoom control box position based on if it's ltr or rtl language
- Move rtl.js function to static/common folder since it's use in browser and in webpack

J=SLAP-1459
TEST=manual

Launched multilang test site
-  see that map appears in en, es, and ar for all map pages (locations, locations_google, locations_full_page_map, locations_full_page_map_with_filters) and providers (Google, Mapbox)
- see that the language displayed in the map matched with the given locale in url
- see that arabic is displayed property on map in rtl styling
- see that zoom control box is in top right for en and es, and is in top left for ar.
